eng/automation/generate_utils.py

```diff
@@ -411,6 +414,9 @@ def generate_typespec_project(
                     # clear existing generated source code, and regenerate
                     drop_changes(sdk_root)
                     remove_generated_source_code(sdk_folder, f"{group_id}.{service}")
+                    _, current_version = set_or_increase_version(sdk_root, group_id, module, version=version)
```

Comment:
Did we call `set_or_increase_version` somewhere else?

If yes, would calling it twice cause problem?

Reply:
> Did we call set_or_increase_version somewhere else?

Yes, we did. 

> would calling it twice cause problem?

No, it wouldn't. 
If version parameter is provided, we [set the version](https://github.com/Azure/azure-sdk-for-java/blob/95fe86d5ca8f596af3e26f0686046e7479689fdd/eng/automation/utils.py#L341) to it, and no further increase. Otherwise, we'll increase version if necessary. Though we never implemented the `increase` logic...

Reply:
Switched back to using `set_or_default_version`, and added `version` parameter to it. 
We may choose to implement `increase` logic, this is trying to avoid further impact...